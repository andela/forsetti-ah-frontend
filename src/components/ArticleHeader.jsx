import React from 'react';
import moment from 'moment';
import { articlePlaceholder, user } from '../assets';
import Button from './common/Button';

const ArticleHeader = ({
  title,
  image,
  username,
  readingTime,
  articleImage,
  createdAt
}) => (
  <div className='container'>
    <div className='row mt-5'>
      <div className='col-md-12 text-left'>
        <h2 className='text-uppercase article-title'>
          { title }
        </h2>
      </div>
    </div>
    <div className='row mt-3 mb-4'>
      <div className='col-3 col-md-1'>
        <img
          src={image !== null || image !== undefined ? image : user}
          alt='profile'
          className='rounded-circle media m-0'
          style={{ objectFit: 'cover', width: '70px', height: '70px' }}
        />
      </div>
      <div className='col-8 col-md-11'>
        <div className='row ml-1'>
          <div className='col-md-12'>
            <p className=''>
              {username}
              <Button type='button' className='btn btn_primary btn-sm py-0 mx-2'> Follow </Button>
            </p>
          </div>
          <div className='col-md-12'>
            <span>
              {
              moment(createdAt).format('DD MMM, YYYY')
            }
            ,
              {
                ` ${Math.ceil(readingTime / 1000.0)} min read`
              }
            </span>
          </div>
        </div>
      </div>
    </div>
    <div className='row'>
      {
      articleImage === null ? <div className='pt-3' />
        : (
          <div className='col-md-12'>
            <img
              src={articleImage !== null ? articleImage : articlePlaceholder}
              alt='articleimage'
              className='col-12 col-md-11 article-image'
            />
          </div>
        )
    }
    </div>
  </div>
);

export default ArticleHeader;
