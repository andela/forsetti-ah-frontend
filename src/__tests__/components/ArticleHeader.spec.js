import React from 'react';
import { shallow } from 'enzyme';
import ArticleHeader from '../../components/ArticleHeader';

describe('Article header component', () => {
  test('should render Article header component correctly', () => {
    const headerProps = {
      title: 'title',
      image: null,
      username: 'username',
      readingTime: 2.5,
      articleImage: null,
      createdAt: '2019-05-14T13:28:50.503Z'
    };
    const wrapper = shallow(<ArticleHeader {...headerProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should render Article header component correctly', () => {
    const headerProps = {
      title: 'title', image: 'image', username: 'username', readingTime: 2.5
    };
    const wrapper = shallow(<ArticleHeader {...headerProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
