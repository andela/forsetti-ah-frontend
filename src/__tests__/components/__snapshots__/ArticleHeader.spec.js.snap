// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article header component should render Article header component correctly 1`] = `ShallowWrapper {}`;

exports[`Article header component should render Article header component correctly 2`] = `ShallowWrapper {}`;
